runners/spark/src/main/java/org/apache/beam/runners/spark/metrics/WithMetricsSupport.java

```diff
@@ -99,21 +97,34 @@ public static WithNamedAggregatorsSupport forRegistry(final MetricRegistry metri
 
   private Map<String, Gauge> extractGauges(final MetricRegistry metricRegistry,
                                            final MetricFilter filter) {
+    Map<String, Gauge> gauges = new HashMap<>();
 
     // find the AggregatorMetric metrics from within all currently registered metrics
-    final Optional<Map<String, Gauge>> gauges =
+    final Optional<Map<String, Gauge>> aggregatorMetrics =
         FluentIterable
             .from(metricRegistry.getMetrics().entrySet())
             .firstMatch(isAggregatorMetric())
-            .transform(toGauges());
+            .transform(aggregatorMetricToGauges());
 
-    return
-        gauges.isPresent()
-            ? Maps.filterEntries(gauges.get(), matches(filter))
-            : ImmutableMap.<String, Gauge>of();
+    // find the SparkBeamMetric metrics from within all currently registered metrics
+    final Optional<Map<String, Gauge>> beamMetrics =
+        FluentIterable
+            .from(metricRegistry.getMetrics().entrySet())
```

Comment:
`metricRegistry.getMetrics().entrySet()` can be extracted to a local variable to avoid duplication.